src/global/project/mod.rs

```diff
@@ -80,31 +105,257 @@ impl Project {
 
-        fs::create_dir_all(&manifest_dir).into_diagnostic()?;
+        tokio::fs::create_dir_all(&manifest_dir)
+            .await
+            .into_diagnostic()
+            .wrap_err_with(|| format!("Couldn't create directory {}", manifest_dir.display()))?;
 
         let manifest_path = manifest_dir.join(MANIFEST_DEFAULT_NAME);
 
         if !manifest_path.exists() {
-            fs::File::create(&manifest_path).into_diagnostic()?;
+            let warn = console::style(console::Emoji("⚠️ ", "")).yellow();
+            let prompt = format!(
+                "{} You don't have a global manifest yet.\n\
+                Do you want to create one based on your existing installation?\n\
+                Your existing installation will be removed if you decide against it.",
+                console::style(console::Emoji("⚠️ ", "")).yellow(),
+            );
+            if !env_root.directories().await?.is_empty()
+                && (assume_yes
+                    || dialoguer::Confirm::new()
+                        .with_prompt(prompt)
+                        .default(true)
+                        .show_default(true)
+                        .interact()
+                        .into_diagnostic()?)
+            {
+                return Self::from_existing_installation(&manifest_path, bin_dir, env_root)
+                    .await
+                    .wrap_err_with(|| {
+                        "Failed to create global manifest from existing installation"
+                    });
+            }
+
+            tokio::fs::File::create(&manifest_path)
+                .await
+                .into_diagnostic()
+                .wrap_err_with(|| format!("Couldn't create file {}", manifest_path.display()))?;
         }
+
         Self::from_path(&manifest_path)
     }
 
+    async fn from_existing_installation(
+        manifest_path: &Path,
+        bin_dir: &BinDir,
+        env_root: &EnvRoot,
+    ) -> miette::Result<Self> {
+        let exposed_binaries: Vec<ExposedData> = bin_dir
+            .files()
+            .await?
+            .into_iter()
+            .filter_map(|path| match is_text(&path) {
+                Ok(true) => Some(Ok(path)), // Success and is text, continue with path
+                Ok(false) => None,          // Success and isn't text, filter out
+                Err(e) => Some(Err(e)),     // Failure, continue with error
+            })
+            .map(|result| result.and_then(Self::exposed_data_from_binary_path))
+            .collect::<miette::Result<_>>()?;
+
+        let parsed_manifest = ParsedManifest::from(exposed_binaries);
+        let toml = toml_edit::ser::to_string(&parsed_manifest).into_diagnostic()?;
+        tokio::fs::write(&manifest_path, &toml)
+            .await
+            .into_diagnostic()?;
+        Self::from_str(manifest_path, &toml)
+    }
+
+    fn exposed_data_from_binary_path(path: PathBuf) -> miette::Result<ExposedData> {
+        let exposed = path
+            .file_stem()
```

Comment:
Good idea. It would be `TryFrom`, instead of `From`.

The disadvantage is that `try_from` doesn't make it clear that the path needs to be the path to the binary. I made a method `ExposedData::from_binary_path` which is also much nicer then just adding it to the `Project`.